x/txfees/keeper/hooks.go

```diff
@@ -14,36 +14,99 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochN
 
 // at the end of each epoch, swap all non-OSMO fees into OSMO and transfer to fee module account
 func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
-	nonNativeFeeAddr := k.accountKeeper.GetModuleAddress(txfeestypes.NonNativeFeeCollectorName)
-	baseDenom, _ := k.GetBaseDenom(ctx)
-	feeTokens := k.GetFeeTokens(ctx)
+	defaultFeesDenom, _ := k.GetBaseDenom(ctx)
 
-	for _, feetoken := range feeTokens {
-		if feetoken.Denom == baseDenom {
+	nonNativeFeeCollectorForStakingRewardsAddr := k.accountKeeper.GetModuleAddress(txfeestypes.NonNativeFeeCollectorForStakingRewardsName)
+	nonNativeFeeCollectorForStakingRewardsBalance := k.bankKeeper.GetAllBalances(ctx, nonNativeFeeCollectorForStakingRewardsAddr)
+
+	// Non-native fee collector for staking rewards get swapped entirely into base denom.
+	for _, coin := range nonNativeFeeCollectorForStakingRewardsBalance {
+		if coin.Denom == defaultFeesDenom {
 			continue
 		}
-		coinBalance := k.bankKeeper.GetBalance(ctx, nonNativeFeeAddr, feetoken.Denom)
+
+		coinBalance := k.bankKeeper.GetBalance(ctx, nonNativeFeeCollectorForStakingRewardsAddr, coin.Denom)
 		if coinBalance.Amount.IsZero() {
 			continue
 		}
 
+		poolId, err := k.protorevKeeper.GetPoolForDenomPair(ctx, defaultFeesDenom, coin.Denom)
```

Comment:
Is it guaranteed that the order does not matter here?

Reply:
Ah nice it looks like you modified to `GetPoolForDenomPairNoOrder` in this commit https://github.com/osmosis-labs/osmosis/pull/6034/commits/72ffbb64d389b83f8b6e22099eb10c1f21670100, thanks :)